boards/itsybitsy_m0/src/lib.rs

```diff
@@ -245,3 +245,16 @@ pub fn usb_allocator(
         usb,
     ))
 }
+
+
+// uf2 bootloaders look for this value to determine if they should stay in the bootloader.
+const ENTER_BOOTLOADER_MAGIC: u32 = 0xf01669ef;
+// The magic value for remaining in the bootloader is stored in the last 4 bytes
+// of RAM. The itsybitsy_m0 has 32k of ram & starts at 0x20000000.
+const ITSYBITSY_BOOTLOADER_MAGIC_ADDR: *mut u32 = (0x20000000 + (32 * 1024) - 4) as *mut _;
```

Comment:
we can add something next to _stack_start in memory.x to point to flash_start or something

Reply:
I dont have a working verison, but memory.x like this would make it less fragile and portable
```
__config__   = ORIGIN(RAM) + LENGTH(RAM) - 4;
SECTIONS
{
  .uf2Segment __config__ : {KEEP(*(.uf2Section))} > RAM
}
```

Then could make it a struct, but then have to init to to something?
maybe DBL_TAP_MAGIC_QUICK_BOOT is the default?
```
pub const DBL_TAP_MAGIC_QUICK_BOOT: u32 = 0xf02669ef;
#[repr(packed)]
pub struct Uf2Config {
    pub magic: u32,
}
#[link_section = ".uf2Section"]
static CONFIG: Uf2Config = Uf2Config {
    magic: DBL_TAP_MAGIC_QUICK_BOOT,
};
```
Or uninit?

```
#[link_section = ".uf2Section"]
static mut CONFIG: MaybeUninit<[u32; 1]> = MaybeUninit::uninit();
pub unsafe fn reset_into_bootloader() {
    CONFIG = MaybeUninit::new([DBL_TAP_MAGIC]);
    cortex_m::peripheral::SCB::sys_reset();
}
```

Reply:
I spent some time trying to get this to work using the linker script, which is made harder by not being allowed custom MEMORY sections. Let me know if you get it working, otherwise would it be okay to stick to mashing pointers at runtime (same way the generated svd code does it?)

Reply:
Not that Im in charge, but yeah I guess theres no other data held there right? So its not buying much.

Reply:
Not in charge either but I trust your intuition as an embedded person / responsible adult lol

Reply:
Heh. Ill take it. My adult says lets get a comment line there saying the 0x20000000 comes from and should be kept up to date with memory.x